Fix 66 Sends all input build details over tube

navidshaikh opened the pull request:
Fixes #66: This commit sends all the input build details to tube.

kbsingh commented:
#dotests

navidshaikh commented on client/build-project.sh:
```diff
@@ -69,9 +68,8 @@ IP=$(ip -f inet addr show eth1 2> /dev/null | grep 'inet' | awk '{ print $2}' |
 
 echo "==> Send build configs to build tube"
-python $CWD/send_build_request.py ${NAME} ${TAG} ${DEPENDS_ON}
+python $CWD/send_build_request.py ${NAME} ${TAG} ${REPO_BRANCH} ${REPO_BUILD_PATH} ${TARGET_FILE} ${NOTIFY_EMAIL} ${DEPENDS_ON}
```
We are sending all the fields from input YAML file for job over tube here.

bamachrn replied:
@navidshaikh Are we planning for change in tube job flow? as these details are only used for build_worker which does not need any of these for now?

navidshaikh replied:
@bamachrn : Not really the flow but the contents passed between different phases of the flow.  Write up for issue https://github.com/CentOS/container-pipeline-service/issues/66#issue-175945472 should help build the context.

navidshaikh commented on client/send_build_request.py:
```diff
@@ -6,10 +6,14 @@
+build_details["repo_build_path"] = sys.argv[4]
+build_details["target_file"] = sys.argv[5]
+build_details["notify_email"] = sys.argv[6]
+build_details["depends_on"] = sys.argv[7]
```
Put all build details over tube.

navidshaikh commented on server/run-delivery.sh:
```diff
@@ -42,7 +42,7 @@ OUTPUT_IMAGE=registry.centos.org/${TARGET_NAMESPACE}/${TO}
 
 _ "Send the image details to test_tube for testing"
-python /tube_request/send_notify_request.py ${BEANSTALK_SERVER} ${OUTPUT_IMAGE} ${NOTIFY_EMAIL}
+python /tube_request/send_notify_request.py ${BEANSTALK_SERVER} ${OUTPUT_IMAGE}
```
No need to explicitly mention the notify email address since it is already in the content present over tube.

dharmit commented on server/send_notify_request.py:
```diff
@@ -7,17 +7,16 @@
-msg_details['msg'] = "Build is successful please pull the image ("+output_image+")"
-msg_details['to_mail'] = notify_email
+msg_details['msg'] = ("Build is successful please "
+                      "pull the image (%s) " % output_image)
```
Since we're removing `msg_details['to_mail'] = notify_email` here, would this fail? Asking sinc we're dumping `msg_details` variable on the `master_tube`. So, the job that's being put on the tube won't have the email details.

navidshaikh replied:
Thanks! I have updated the PR.

bamachrn commented:
#dotests

bamachrn commented:
@navidshaikh do you want to rebase this?

dharmit commented:
I'm in favor of testing this out outside the CI before merging into master. Just to avoid breaking things. Because, this PR is modifying a lot of pieces.

Thoughts? @kbsingh @bamachrn @navidshaikh

navidshaikh commented:
> @navidshaikh do you want to rebase this?

Done!

navidshaikh commented:
> I'm in favor of testing this out outside the CI before merging into master. Just to avoid breaking things. Because, this PR is modifying a lot of pieces.

+1

Would you like to suggest some particular cases / setup, to test this out?

dharmit commented:
@navidshaikh I'd like us to setup a separate environment where we start the service based off the code in this PR and throw a few builds its way. My major concerns are:

- inter service communication: since we're adding all the possible keys to json at the beginning, we need to make sure that no service is doing something by itself and breaking things.
- delivery/notification: as I commented on the code, I'd like to see that the images get built and notifications get sent out correctly.
- test a bit longer: recently things broke somewhat badly. Testing this by building a few images and checking everything from *build* to *scan* to *deliver/notify is working well* is important.

What do you suggest @bamachrn @kbsingh?

kbsingh commented:
@dharmit sure, would this be a devcloud instance ?

dharmit commented:
> @dharmit sure, would this be a devcloud instance ?

Sure! In complete favor of using a c7.large system and use it as an all-in-one setup.

navidshaikh commented:
Rebased the PR with master, added a new commit replacing all the occurrences of variable names `to_email` and `to_mail` with `notify_email` to have consistency.

bamachrn commented:
I would vote for four VM setup as we can also test the scanner part there.

dharmit commented:
From code point of view, LGTM :+1: 

Would request pep8, etc. changes to not be mixed with features/bugs like in this PR. A separate PR for those makes more sense.

navidshaikh commented:
@bamachrn , @rtnpro : can you please code review this PR ?

bamachrn commented on server/send_scan_request.py:
```diff
@@ -7,6 +7,7 @@
 print "Getting build details from jenkins"
 beanstalk_host = sys.argv[1]
 image_details = {}
+image_details['action'] = "start_test"
```
should this be not changed to start_scan instead of start_test?

bamachrn commented:
LGTM except that action name. I believe it should be changed to start_scan.

dharmit commented:
LGTM :+1:

mohammedzee1000 commented:
LGTM  :+1:
